### PR TITLE
Add default name to commands for lazy loading

### PR DESCRIPTION
**Pull request by bennothommo**
Refs:
- https://github.com/laravel/framework/pull/34873
- https://github.com/wintercms/winter/pull/148

### CODE REVIEW

**Comment by LukeTowers** [comment]
What needs to be done in order to support this for plugin registered commands?

**Comment by bennothommo** [comment]
@LukeTowers I think if they define `$defaultName` as a property with the commands, the lazy loading will kick in automatically. At the very least, I don't believe they've disabled eager-loading commands in the old fashion.

File: src/Scaffold/Console/command/command.stub [hunk 1]
```diff
@@ -9,7 +9,7 @@ class {{studly_name}} extends Command
     /**
      * @var string The console command name.
      */
-    protected $name = '{{lower_plugin}}:{{lower_name}}';
+    protected static $defaultName = '{{lower_plugin}}:{{lower_name}}';
```
Comment (LukeTowers) [inline src/Scaffold/Console/command/command.stub:12]: I think we should probably include the signature property like Laravel is doing in their scaffolding

File: src/Scaffold/Console/CreateTheme.php [hunk 2]
```diff
@@ -8,11 +8,11 @@
...
 {
     /**
-     * The console command name.
+     * The default command name for lazy loading.
      *
-     * @var string
+     * @var string|null
      */
-    protected $name = 'create:theme';
+    protected static $defaultName = 'create:theme';
```
Comment (LukeTowers) [inline src/Scaffold/Console/CreateTheme.php:13]: What was the name property being used by previously and are we save to no longer use it?

File: src/Scaffold/Console/command/command.stub (hunk 1 above)
Comment (bennothommo) [reply src/Scaffold/Console/command/command.stub:12]: Sounds good.

File: src/Scaffold/Console/CreateTheme.php (hunk 2 above)
Comment (bennothommo) [reply src/Scaffold/Console/CreateTheme.php:13]: I believe `$name` is used in the absence of a `$signature`. It's safe to remove if we use `$defaultName` or `$signature`, as far as I can tell.

**Pull request merged**